lib/charms/opensearch/v0/opensearch_nodes_exclusions.py

```diff
@@ -44,73 +36,54 @@ def __init__(self, charm):
 
         self._scope = Scope.APP if self._charm.unit.is_leader() else Scope.UNIT
 
-    def add_current(self) -> None:
-        """Add Voting and alloc exclusions."""
-        if (self._node.is_cm_eligible() or self._node.is_voting_only()) and not self._add_voting():
-            logger.error(f"Failed to add voting exclusion: {self._node.name}.")
-
-        if self._node.is_data() and not self._add_allocations():
-            logger.error(f"Failed to add shard allocation exclusion: {self._node.name}.")
-
-    def delete_current(self) -> None:
-        """Delete Voting and alloc exclusions."""
-        if (
-            self._node.is_cm_eligible() or self._node.is_voting_only()
-        ) and not self._delete_voting():
-            self._charm.peers_data.put(self._scope, VOTING_TO_DELETE, True)
-
-        if self._node.is_data() and not self._delete_allocations():
-            current_allocations = set(
-                self._charm.peers_data.get(self._scope, ALLOCS_TO_DELETE, "").split(",")
-            )
-            current_allocations.add(self._node.name)
-
-            self._charm.peers_data.put(
-                self._scope, ALLOCS_TO_DELETE, ",".join(current_allocations)
-            )
-
-    def cleanup(self) -> None:
-        """Delete all exclusions that failed to be deleted."""
-        need_voting_cleanup = self._charm.peers_data.get(self._scope, VOTING_TO_DELETE, False)
-        if need_voting_cleanup and self._delete_voting():
-            self._charm.peers_data.delete(self._scope, VOTING_TO_DELETE)
-
+    def allocation_cleanup(self) -> None:
+        """Delete alloc exclusions that failed to be deleted."""
         allocations_to_cleanup = self._charm.peers_data.get(
             self._scope, ALLOCS_TO_DELETE, ""
         ).split(",")
-        if allocations_to_cleanup and self._delete_allocations(allocations_to_cleanup):
+        if allocations_to_cleanup and self.delete_allocations_exclusion(allocations_to_cleanup):
```

Comment:
nit: adding `_exclusions` at the end is redundant, given that we are already in the "exclusions" module.